nvflare/app_common/dex/fl_model_exchanger.py

```diff
@@ -0,0 +1,77 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Optional, Tuple
+
+from nvflare.app_common.abstract.fl_model import FLModel
+from nvflare.app_common.dex.dxo_exchanger import DXOExchanger
+from nvflare.app_common.utils.fl_model_utils import FLModelUtils
+
+
+class FLModelExchanger:
+    def __init__(self, exchanger: DXOExchanger):
+        self.exchanger = exchanger
+
+    def send_request(self, model: FLModel, timeout: Optional[float] = None) -> Tuple[bool, str]:
```

Comment:
No, this is not the highest-level API.  This API is still part of the "basic mechanism". User-facing APIs need to be built on top of this.

The destination is always "the other side", which is configured with "pipe_role" (or side) and "pipe_name", which is a path (file based) or URL (IPC based) or queue id (memory based).

send_request is to send the FLModel object to the peer.  Pipe is just the underlying mechanism for transmitting the data and it doesn't matter whether it's file based or not.

I guess we could send FLModel directly without having to convert to DXO, because the underlying pipe doesn't really care what type of data it is. In fact, if you look at DXOExchanger, there is no DXO-specific processing at all. So it could be simply called DataExchanger, and the type of input/output should be type "Any" (replace DXO with Any). The place that serialize the payload is in FileAccessor object.  We won't need dxo_file_exchanger any more.

The DataExchanger could be used for any data, and the pipe takes care of converting the data to/from the right data objects (e.g. FLModel). I like this!

Then this fl_model_exchanger (or a different name) could be re-purposed to be the user-facing API. For the ease of adoption, perhaps we should have two sub-APIs: one for the sender side (FLARE side), one for the external app side. FLARE side only cares about send_model and get_reply; external app cares about get_model, submit_metrics, and submit_model.